integration/run.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import argparse
+import os
+import re
+import shutil
+import subprocess
+
+import structlog
+import yaml
+from taskcluster.helper import TaskclusterConfig
+
+taskcluster = TaskclusterConfig("https://community-tc.services.mozilla.com")
+
+logger = structlog.getLogger(__name__)
+
+ROOT = os.path.realpath(os.path.dirname(__file__))
+
+
+def configure():
+    """
+    Load configuration from CLI args and Taskcluster secrets
+    """
+    parser = argparse.ArgumentParser(description="Run code-review integration tests")
+    parser.add_argument(
+        "-c",
+        "--configuration",
+        help="Local configuration file replacing Taskcluster secrets",
+        type=open,
+    )
+    parser.add_argument(
+        "--clone-dir",
+        help="Directory where to clone repositories",
+        default=os.environ.get("CLONE_DIR", os.path.join(ROOT, "clone")),
+    )
+    parser.add_argument(
+        "--taskcluster-secret",
+        help="Taskcluster Secret path",
+        default=os.environ.get("TASKCLUSTER_SECRET"),
+    )
+    args = parser.parse_args()
+
+    taskcluster.auth()
+    taskcluster.load_secrets(
+        args.taskcluster_secret,
+        required=("phabricator", "admins"),
+        existing={"admins": ["babadie@mozilla.com"]},
+        local_secrets=yaml.safe_load(args.configuration)
+        if args.configuration
+        else None,
+    )
+
+    # Make sure the clone dir is available
+    os.makedirs(args.clone_dir, exist_ok=True)
+
+    return args
+
+
+def clone(url, directory, branch="tip"):
+    """
+    Mercurial clone with robustcheckout
+    """
+    logger.info("Cloning repository", url=url, dir=directory)
+
+    # Parent should exist, not current dir
+    assert os.path.exists(os.path.dirname(directory)), "Missing parent of clone dir"
+
+    # Cleanup existing target
+    if os.path.exists(directory):
+        logger.info("Removing previous clone")
+        shutil.rmtree(directory)
+
+    # Now let's clone
+    cmd = [
+        "hg",
+        "robustcheckout",
+        "--purge",
+        f"--sharebase={directory}-shared",
+        f"--branch={branch}",
+        url,
+        directory,
+    ]
+    subprocess.check_output(cmd)
+
+
+def tip(repo_dir):
+    """
+    Get the tip of the repo
+    """
+    cmd = ["hg", "tip", "--template={rev}"]
+    rev = subprocess.check_output(cmd, cwd=repo_dir)
+    return int(rev)
+
+
+def patch(filename, repo_dir, message):
+    """
+    Apply a locally stored patch on the repository
+    and commit the difference
```

Comment:
The main painful thing will be to keep the patches updated. We should patch files which are rarely changing.